itest/tests/preferences.test.js

```diff
@@ -0,0 +1,28 @@
+/* @flow */
+import path from "path"
+
+import {
+  defaultModalButton,
+  ingestProgressBar,
+  jsonTypeConfigInput,
+  viewerResults
+} from "../../src/js/test/locators"
+import {stdTest} from "../lib/jest"
+import createTestBrim from "../lib/createTestBrim"
+
+const config = path.join(__dirname, "../testdata/custom-schema.json")
+
+describe("Preferences Modal Tests", () => {
+  const brim = createTestBrim("preferences.test")
+
+  stdTest("setting a custom json type config", async (done) => {
+    brim.clickAppMenuItem("preferences")
+    await brim.setValue(jsonTypeConfigInput, config)
+    await brim.click(defaultModalButton)
+    await brim.ingest("custom-sample.ndjson")
+    await brim.waitUntil(() => brim.isNotVisible(ingestProgressBar))
```

Comment:
In general, tests shouldn't be asked to do waiting. That is better served with application steps. This particular pattern is already done with `ingestFile()`. You could add another method to createTestBrim for that, right?

Reply:
Oh, ingest file already handles waiting for the ingest to finish. `brim.ingest` internally uses `ingestFile`, so this line is not needed.